# Association Rule Using Apriori Algorithm
Here in this notebook we have taken a dataset containing 7501 rows each row contain a transaction each column represent a particular
item in transaction then we applied apriori alogrithm to find association between item this dataset is  belongs to a supermarket from france 

Importing libraries

In [15]:
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

importing dataset using read_csv method from pandas

In [16]:
dataset = pd.read_csv('Market_Basket_Optimisation.csv', header = None)

Have a look at dataset here we can see each column contains a items in transaction and each row represnts a single transaction

In [24]:
dataset.head()

0          1           2   ...               17       18         19
0         shrimp    almonds     avocado  ...  frozen smoothie  spinach  olive oil
1        burgers  meatballs        eggs  ...              NaN      NaN        NaN
2        chutney        NaN         NaN  ...              NaN      NaN        NaN
3         turkey    avocado         NaN  ...              NaN      NaN        NaN
4  mineral water       milk  energy bar  ...              NaN      NaN        NaN

[5 rows x 20 columns]

Here we have converted each row into vector and added into transactions list making it matrix 

transactions = []
for i in range(0, 7501):
    transactions.append([str(dataset.values[i,j]) for j in range(0, 20)])

Here we have imported module and create apriori model with minimium support of 0.003 and minimium confidence of 0.2 and minimium lift of 3 and minimiun length of two

In [19]:
from apyori import apriori
rules = apriori(transactions, min_support = 0.003, min_confidence = 0.2, min_lift = 3, min_length = 2)

here we have created rules and printed all rules with above specifications

In [20]:
results = list(rules)
for i in results:
    print(i)

RelationRecord(items=frozenset({'light cream', 'chicken'}), support=0.004532728969470737, ordered_statistics=[OrderedStatistic(items_base=frozenset({'light cream'}), items_add=frozenset({'chicken'}), confidence=0.29059829059829057, lift=4.84395061728395)])
RelationRecord(items=frozenset({'escalope', 'mushroom cream sauce'}), support=0.005732568990801226, ordered_statistics=[OrderedStatistic(items_base=frozenset({'mushroom cream sauce'}), items_add=frozenset({'escalope'}), confidence=0.3006993006993007, lift=3.790832696715049)])
RelationRecord(items=frozenset({'escalope', 'pasta'}), support=0.005865884548726837, ordered_statistics=[OrderedStatistic(items_base=frozenset({'pasta'}), items_add=frozenset({'escalope'}), confidence=0.3728813559322034, lift=4.700811850163794)])
RelationRecord(items=frozenset({'honey', 'fromage blanc'}), support=0.003332888948140248, ordered_statistics=[OrderedStatistic(items_base=frozenset({'fromage blanc'}), items_add=frozenset({'honey'}), confidence=0.245098

from above solution it can be interpreted that the first rule is light cream with chicken it has confidence of 29% and second 
rule is esclope with mushroom cream sauce with confidence of 30% likewise we can interpret all rules although first five to ten rules of more significance as there are more chance of them to occur